airflow/providers/elasticsearch/hooks/elasticsearch.py

```diff
@@ -0,0 +1,89 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from es.elastic.api import connect
+
+from airflow.hooks.dbapi_hook import DbApiHook
+
+
+class ElasticsearchHook(DbApiHook):
+    """
+        Interact with Elasticsearch through the elasticsearch-dbapi
+    """
+
+    conn_name_attr = 'elasticsearch_conn_id'
+    default_conn_name = 'elasticsearch_default'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.schema = kwargs.pop("schema", None)
+        self.connection = kwargs.pop("connection", None)
```

Comment:
Would it be better to make those arguments with default values `None`?

Reply:
about **schema** it's better to set it to be **http**  because by default the **elasticsearch** connection Uri starts with `elasticsearch+http` and the default values in `elasticsearch_default` schema is **http**, I can set  it **http** instead **None** in the code, I mean
```
self.schema = kwargs.pop("schema", "http")
```
about the connection, I have no idea about it. I just checked my code with other providers hooks like as presto

Reply:
I would suggest the following:
```python
def __init__(
    self, 
    schema: str = "http",
    connection: str = "elasticsearch_conn_id, 
    *args, 
    **kwargs
):
    super().__init__(*args, **kwargs)
    self.schema = schema
    self. connection = connection
```
It's definitely a better pattern that poping arguments from kwargs. Especially if someone will use hook passing args instead of kwargs ;)

Reply:
Okay. I will do it now.

Reply:
Done!

Reply:
Actually, `self.connection` can not be as string and I did this:
```
 def __init__(self,
                 schema: str = "http",
                 connection=None,
                 *args,
                 **kwargs):
        super().__init__(*args, **kwargs)
        self.schema = schema
        self.connection = connection
```